src/main/java/org/candlepin/subscriptions/security/IdentityHeaderAuthenticationDetailsSource.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright (c) 2009 - 2019 Red Hat, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * Red Hat trademarks are not licensed under GPLv3. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+
+package org.candlepin.subscriptions.security;
+
+import static org.candlepin.subscriptions.security.IdentityHeaderAuthenticationFilter.*;
+
+import org.candlepin.subscriptions.exception.ErrorCode;
+import org.candlepin.subscriptions.exception.SubscriptionsException;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.security.authentication.AuthenticationDetailsSource;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.authority.mapping.Attributes2GrantedAuthoritiesMapper;
+import org.springframework.security.web.authentication.preauth.PreAuthenticatedGrantedAuthoritiesWebAuthenticationDetails;
+import org.springframework.util.StringUtils;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Base64;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Map;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.ws.rs.core.Response;
+
+/**
+ * Class in charge of populating the security context with the users roles based on the values in the
+ * x-rh-identity header.
+ */
+public class IdentityHeaderAuthenticationDetailsSource implements
+    AuthenticationDetailsSource<HttpServletRequest,
+    PreAuthenticatedGrantedAuthoritiesWebAuthenticationDetails> {
+
+    private static final Logger log =
+        LoggerFactory.getLogger(IdentityHeaderAuthenticationDetailsSource.class);
+
+    // NB: Right now there are just two roles.  If we expand on that, we should externalize all the
+    //  roles to an enum, serialize the header JSON to an object, and write an actual predicate method
+    //  isUserInRole
+    private static final String ORG_ADMIN = "ORG_ADMIN";
+    private static final String INTERNAL = "INTERNAL";
```

Comment:
Right now they aren't used for anything. :laughing: But I thought we should at least read them in as roles for potential future use.  I can comment on that.

Reply:
Actually, I'm going to leave out the comment since I don't want us to start using them and then have the comment out of date.  I think it's fairly easy to determine that, yeah these roles are populated but we don't ever check them for anything because we don't have anything that requires them yet.